storage/mysql/mysql.go

```diff
@@ -234,16 +235,16 @@ func (s *Storage) writeTreeState(ctx context.Context, tx *sql.Tx, size uint64, r
 	return nil
 }
 
-// ReadTile returns a full tile or a partial tile at the given level, index and width.
-// If the tile is not found, nil is returned with no error.
+// ReadTile returns a full tile or a partial tile at the given level, index and treeSize.
+// If the tile is not found, it returns os.ErrNotExist.
 //
 // TODO: Handle the following scenarios:
 // 1. Full tile request with full tile output: Return full tile.
 // 2. Full tile request with partial tile output: Return error.
 // 3. Partial tile request with full/larger partial tile output: Return trimmed partial tile with correct tile width.
 // 4. Partial tile request with partial tile (same width) output: Return partial tile.
 // 5. Partial tile request with smaller partial tile output: Return error.
-func (s *Storage) ReadTile(ctx context.Context, level, index, width uint64) ([]byte, error) {
+func (s *Storage) ReadTile(ctx context.Context, level, index, minTreeSize uint64) ([]byte, error) {
```

Comment:
I think some of these TODOs can be marked TODONE, just 3 and 4 to ponder?

Reply:
The logic now combines both `2` and `5` together. Do we want to return different error?

Reply:
Ah, please ignore my comment. My assumption was wrong. The tree size is provided by the user but not the internal tree size.

Reply:
Thanks!